modules/lang-painless/spi/src/main/java/org/elasticsearch/painless/spi/WhitelistField.java

```diff
@@ -37,10 +41,17 @@
     /** The canonical type name for the field which can be used to look up the Java field through reflection. */
     public final String canonicalTypeNameParameter;
 
+    /** The {@link Map} of annotations for this field. */
+    public final Map<Class<?>, PainlessAnnotation> painlessAnnotations;
+
     /** Standard constructor.  All values must be not {@code null}. */
-    public WhitelistField(String origin, String fieldName, String canonicalTypeNameParameter) {
+    public WhitelistField(String origin, String fieldName, String canonicalTypeNameParameter,
+            Map<Class<?>, PainlessAnnotation> painlessAnnotations) {
```

Comment:
I think we should pass in a list of annotations, and build the map ourselves internally. This way a user manually constructing a whitelist (not using the parser) can't mess up the map.

Reply:
Done.